assets/src/hooks/v2/use_interval_paging.tsx

```diff
@@ -0,0 +1,64 @@
+import { useState, useEffect, useRef, useCallback } from "react";
+
+interface UseIntervalPagingProps {
+  numPages: number; // Total number of pages to cycle through
+  cycleIntervalMs?: number; // In milliseconds, the interval for cycling pages (only used if advanceOnDataRefresh is false)
+  onFinish: () => void; // Callback when cycling completes
+}
+
+/**
+ * This hook acts as a way for components with multiple pages to advance
+ * through their pages on a set time interval.
+ */
+function useIntervalPaging({
+  numPages,
+  cycleIntervalMs,
+  onFinish,
+}: UseIntervalPagingProps) {
+  const [pageIndex, setPageIndex] = useState(0);
+  // Use refs to keep stable references to numPages and interval without triggering re-renders
+  const intervalRef = useRef<NodeJS.Timeout | null>(null);
+  const numPagesRef = useRef(numPages);
+
+  useEffect(() => {
+    numPagesRef.current = numPages;
```

Comment:
Based on the current usage, it looks like we don't expect this value to change over the lifecycle of the component, so we could probably just use the argument directly without storing it in a ref.

Reply:
Actually, looking at this some more, I'm less sure about this. If this does change, what's the expected behavior?

Reply:
`numPages` could change over the lifecycle of a component. We won't update the data being displayed in the component though until we reach page 0 after a refresh has been triggered during paging. This ref was necessary with the old functionality of a hook handling both refresh/interval paging, but I think I can get this working without a ref now.